packages/ui/src/components/va-tree-view/VaTreeView.demo.vue

```diff
@@ -1,59 +1,266 @@
 <template>
   <VbDemo>
-    <VbCard>
-      <va-button @click="$refs.treeView.expand()">
-        EXPAND ALL
-      </va-button>
-      <va-button @click="$refs.treeView.collapse()">
-        COLLAPSE ALL
-      </va-button>
-
-      <va-tree-view ref="treeView">
-        <va-tree-category
-          label="Category text"
-          is-open
-        >
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-          <va-tree-category
-            label="Category text"
-            is-open
-          >
-            <va-tree-node>
-              Node text
-            </va-tree-node>
-          </va-tree-category>
-        </va-tree-category>
+    <VbCard title="Default">
+      <va-tree-view :nodes="nodes" />
+    </VbCard>
+    <VbCard title="Icons">
+      <va-tree-view :nodes="nodesWithIcons" />
+      <va-tree-view :nodes="nodesWithIcons">
+        <template #icon="node">
+          <va-icon
+            :name="node.icon"
+            :color="node.iconColor"
+            :title="node.label"
+            size="20px"
+          />
+        </template>
       </va-tree-view>
     </VbCard>
-    <VbCard title="Color">
+    <VbCard title="Selection and icons">
+      <va-radio
+        v-for="(option, index) in selectionTypeOptions"
+        :key="index"
+        v-model="selectionType"
+        :option="option"
+      />
+      <br />
+      <va-color-palette
+        v-model="selectedColor"
+        :palette="colorsPalette"
+      />
+      <br />
+      <p>
+        {{ JSON.stringify(selectedNodes) }}
+      </p>
+      <br />
+      <va-tree-view
+        v-model:checked="selectedNodes"
+        :nodes="nodesWithIcons"
+        :color="selectedColor"
+        :selection-type="selectionType"
+        expand-all
```

Comment:
Let's split these into multiple demos, so that each demo covers one piece of functionality. This one does too much.